src/team184/RobotPlayer.java

```diff
@@ -21,181 +21,446 @@
+                            visited.add(test);
+                        }
+                        toAdd.cameFrom = curr;
+                        if (toAdd.mannhattan() < minCost) {
+                            minCost = toAdd.mannhattan();
+                            minAction = toAdd;
+                        }
+                        front.add(toAdd);
+                    }
+                } catch (GameActionException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+        }
+        Stack<Action> moves = new Stack<Action>();
+        while (minAction.cameFrom != null) {
+            moves.push(minAction);
+            minAction = minAction.cameFrom;
+        }
+        return moves;
     }
-    
-    public static void move(Action action, RobotController rc){
-    	System.out.println(action);
-    	if(action.type == MyActionType.DIG){
-    		try {
-				rc.clearRubble(rc.getLocation().directionTo(action.location));
-			} catch (GameActionException e) {
-				// TODO Auto-generated catch block
-				e.printStackTrace();
-			}
-    	}
-    	if(action.type == MyActionType.MOVE){
-    		try {
-    			if(rc.canMove(rc.getLocation().directionTo(action.location)))
-					rc.move(rc.getLocation().directionTo(action.location));
-			} catch (GameActionException e) {
-				// TODO Auto-generated catch block
-				e.printStackTrace();
-			}
-    	}
+
+    public static void move(Action action, RobotController rc) {
+        System.out.println(action);
+        if (action.type == MyActionType.DIG) {
+            try {
+                rc.clearRubble(rc.getLocation().directionTo(action.location));
+            } catch (GameActionException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+        if (action.type == MyActionType.MOVE) {
+            try {
+                if (rc.canMove(rc.getLocation().directionTo(action.location)))
+                    rc.move(rc.getLocation().directionTo(action.location));
+            } catch (GameActionException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
     }
-    
+
+    @SuppressWarnings("unused")
     public static void run(RobotController rc) {
-    	Direction[] directions = {Direction.NORTH, Direction.NORTH_EAST, Direction.EAST, Direction.SOUTH_EAST,
-                Direction.SOUTH, Direction.SOUTH_WEST, Direction.WEST, Direction.NORTH_WEST};
-    	RobotType[] robotTypes = {RobotType.SCOUT, RobotType.SOLDIER, RobotType.SOLDIER, RobotType.SOLDIER,
-                RobotType.GUARD, RobotType.GUARD, RobotType.VIPER};
-                    
-    	Stack<Action> moves = null;
+        Stack<Action> moves = null;
         if (rc.getType() == RobotType.ARCHON) {
+            runARCHON(rc, moves);
+        } else if (rc.getType() != RobotType.ARCHON) {
+            runSCOUT(rc, moves);
```

Comment:
I think these should be in a switch statement, or else if block. Otherwise they'll all get run for one robot